File: 프로그래머스/3/12987. 숫자 게임/숫자 게임.py
from collections import deque

def solution(A, B):
    A.sort()
    B.sort()
    
    A = deque(A)
    B = deque(B)
    
    cnt = 0
    
    while A and B:
        if A[0] < B[0]:
            cnt += 1
            A.popleft()
            B.popleft()
        else :
            B.popleft()
    
    return cnt

print(solution([1,3,9,9], [2,2,8,10])) # 3나와야함

File: 프로그래머스/3/12987. 숫자 게임/숫자 게임.js
function solution(A, B) {
    A.sort((a,b) => a - b);
    B.sort((a,b) => a - b);
    
    let ai = 0;
    let bi = 0;
    let cnt = 0;
    while (ai < A.length && bi < B.length){
        if (A[ai] < B[bi]){
            ai++;
            bi++;
            cnt++;
        }
        else {
            bi++;
        }
    }
    
    return cnt;
}